File: module_base/src/main/java/com/yyxnb/module_base/base/BaseData.java
package com.yyxnb.module_base.base;

import com.yyxnb.common.interfaces.IData;

public class BaseData<T> implements IData<T> {

    public String status;
    public String message;
    public T data;

    @Override
    public String getCode() {
        return status;
    }

    @Override
    public String getMsg() {
        return message;
    }

    @Override
    public T getResult() {
        return data;
    }

    @Override
    public boolean isSuccess() {
        return "200".equals(status);
    }
}

File: common_base/src/main/java/com/yyxnb/common_base/base/BaseData.kt
package com.yyxnb.common_base.base

import com.yyxnb.widget.interfaces.IData

data class BaseData<T>(
        var status: String = "",
        var message: String = "",
        var data: T
) : IData<T> {

    override fun getCode(): String = status

    override fun getMsg(): String = message

    override fun getResult(): T? = data

    override fun isSuccess(): Boolean = "200" == status

    override fun id(): Int {
        return hashCode()
    }
}